projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/FileUtils.java

```diff
@@ -100,14 +100,17 @@ public static File unzipDataJob(Resource resource, File tempDir, String jobName)
     File jobZipFile = new File(tempDir, "job_zip.zip");
     File jobTempDir = new File(tempDir, "job_dir");
     org.apache.commons.io.FileUtils.copyInputStreamToFile(resource.getInputStream(), jobZipFile);
-    new ZipFile(jobZipFile).extractAll(jobTempDir.getAbsolutePath());
+    try (ZipFile theZipFile = new ZipFile(jobZipFile)) {
```

Comment:
so ZipFile close implementation would handle openInputStreams and there are none, so no actual resources leak is patched. However it's a good style to close Closable objects upon usage, due to in the future implementation may change.

Reply:
Hey @ivakoleva , 
Could you elaborate please? The theZipeFile object is declared in a try-with-resource block and is going to be automatically closed.

Reply:
the PR is named "potential resource leak fixes", I look to understand are there any leaks or not? This helps classify the old release severity, new release adoption and prioritization of adoption/fix. 
The ZipFile implements Closable, so it should be closed. In terms of severity, there was no actual resource leak.

Reply:
ZipFile also implements AutoClosable, which allows the class to be used in try-with-resource block and after the block is finished, the JVM will automatically close it. 
This change is to address a Coverity scan warning for a _potential_ resource leak which could occur if the usage of the ZipFile object causes an exception which was previously not handled in this method.

Reply:
The automated tools could hint you where to look;